odl/tomo/backends/astra_cpu.py

```diff
@@ -104,26 +105,47 @@ def astra_cpu_forward_projector(vol_data, geometry, proj_space, out=None):
     proj_id = astra_projector(vol_interp, vol_geom, proj_geom, ndim,
                               impl='cpu')
 
-    # Create ASTRA data structures
-    vol_id = astra_data(vol_geom, datatype='volume', data=vol_data,
-                        allow_copy=True)
+    if vol_data.space.is_rn:
```

Comment:
Strongly disagree with this implementation for two reasons:

* Double the implementation effort
* Double the memory overhead at runtime

A better way would (imo) be to do all of this inside the main "RayTransform" class and call the backends with real arrays in all cases.